lib/reek/smell_detectors/base_detector.rb

```diff
@@ -122,6 +124,25 @@ def valid_detector?(detector)
           descendants.map { |descendant| descendant.to_s.split('::').last }.
             include?(detector)
         end
+
+        #
+        # @param detector_name [String] the detector in question, e.g. 'DuplicateMethodCall'
+        # @return [SmellDetector] - this will return the class, not an instance
+        #
+        def to_detector(detector_name)
+          SmellDetectors.const_get detector_name
+        end
+
+        #
+        # @return [Set<Symbol>] - all configuration keys that are available for this detector
+        #
+        def configuration_keys
+          BASIC_OPTIONS + custom_options
```

Comment:
`default_options` is the method each smell detector implements containing option keys and their default values. Since the goal here is to get those keys in symbolized form, it seems most straightforward to take them straight from `default_options`.